### ISSUE TITLE
Multiplayer support

### ISSUE THREAD

**Issue by engeplabs** [zlsa/atc#392]
I was thinking of this when I first began to play around with this atc simulator. At some point I just couldn't handle all the planes at Kennedy Int. so I would have been very happy for having a helping hand. So I would like to introduce multiplayer to this sim.

I thougt of letting the game run on the server and sending inputs from clients to the server plus sending the gamestate from the server back to the clients. Also it could be possible to have a peer to peer connection between two clients and have one of them acting as the host for the game.

This is just an idea and may need some more investigation on the possiblities we have with the code as it is in the current state. But I would be very interested and motivated to help to develop this kind of feature.

So what do you guys and girls think? :grin:

**Comment by Maverick283** [comment]
Just recently had this suggestion (check out #359) but decided to postpone this as there is other stuff to do right now... Once the basis is there it will for sure be considered to add a multiplayer feature.

**Comment by sixeyeco** [comment]
:+1:  Your proposal sounds great! There's already been positive feedback to the idea, the latest post was this one: https://github.com/zlsa/atc/issues/359

Is this something you could do on your fork? I would love to help. I can't write code but I'd be available for testing!

**Comment by engeplabs** [comment]
@Maverick283 I'm very sorry for posting this issue then. This happens, when you are just to stupid to use the search function... :see_no_evil::wink:
Should I close/delete this issue then?
I am pretty new to all this github type of thing but I really would like to participate in this project. I guess I should just make a pull request and get working? Also you said that there are open issues which have to be solved first.. how does the priority kind of stuff work in this project? Are there some guidelines?

@sixeyeco Yes, I probably could do it. And yeah testing is always the fun part when it comes to games isn't it? :stuck_out_tongue_winking_eye: But of course, when I have some kind of solution I will come back to you. :relieved:

**Comment by Maverick283** [comment]
No sure, get yourself a fork of this project and start working... :D  
Well due to the limited set of programmers (or rather those who know JS) it's pretty much up to the programmers to set priorities... If you get a working multiplayer code going and it can be merged into the main project, I as of now see no reason why it wouldn't be merged!

**Comment by engeplabs** [comment]
Ok great! ^^
I think I will try to tackle down one of the minor issues before starting with the implementation of the multiplayer features, just to get a hang of this git thingy.
And who has an idea of JS if I might ask? Because my knowledge of JS only contains some basic stuff. So it would be good if I could sync with somebody if I get stuck.

**Comment by erikquinn** [comment]
@patitoxd I am _NOT_ an expert with js, for that I'd look more toward some of the guys who have been around longer. @tedrek is probably the most active guy who _actually_ knows what he's doing, unlike me (my usefulness mostly comes in my atc expertise), but I am happy to lend a hand if it means the sim gets better! This is quite an active group, so help and feedback will come easily from many people.

**Comment by MaicomMR** [comment]
Hey guys and girls, I like a lot and support your idea, but i not have a good knowledge of Java language, but I work with Photoshop and Illustrator, at need anything that involves this call me :+1:

**Comment by tedrek** [comment]
Better late to the thread than never... we had a new baby which explains my absence.

I certainly support the concept of multiplayer.  I see many technical hurdles to implementing it, none of which are insurmountable. @patitoxd, @erikquinn, I am happy to help with Javascript and a number of other languages.  I may also be able to provide some server resources.

I would love to hear any technical proposals for extending the game to support multi-player.

**Comment by erikquinn** [comment]
@tedrek Oh wow, congratulations! :smiley: 

I am also very willing to help on this, but have zero clue about how that sort of thing is done.

**Comment by erikquinn** [comment]
The ATC repository is being migrated to it's new home at [https://github.com/openscope/openscope](https://github.com/openscope/openscope), 
and thus, all issues are being closed. If this is still an issue with the latest version of the sim 
(accessible at [http://www.openscope.co](http://www.openscope.co)), or is a feature you still think we are lacking, 
please reopen the issue at the new repo.

*Please note that the vast majority of these issues have been copied to the new repository, or else are covered by other issues created there. See the below screenshot for what it looks like when your issue is known in the new repo:*

![image](https://cloud.githubusercontent.com/assets/5103735/22630816/5a32d794-ebce-11e6-9c46-d61a262e2b23.png)


 Thank you!


Closing this issue.
